src/agora/radio_calibrate.cc

```diff
@@ -557,39 +557,42 @@ void RadioConfig::AdjustDelays(std::vector<int> offset) {
 }
 
 bool RadioConfig::InitialCalib(bool sample_adjust) {
-  size_t seq_len = cfg_->PilotCf32().size();
+  size_t seq_len = cfg_->PilotCf32().size();  // excludes zero padding
   size_t read_len = cfg_->PilotCi16().size();
 
   // Transmitting from only one chain, create a null vector for chainB
   std::vector<std::complex<int16_t>> dummy_ci16(read_len, 0);
 
-  size_t ch = cfg_->Channel() == "B" ? 1 : 0;
   std::vector<void*> txbuff0(2);
+  std::vector<void*> txbuff1(2);
   txbuff0[0] = cfg_->PilotCi16().data();
+  if (cfg_->NumChannels() == 2) {
+    std::vector<std::complex<int16_t>> zeros(read_len,
+                                             std::complex<int16_t>(0, 0));
+    txbuff0[1] = zeros.data();
+    txbuff1[0] = zeros.data();
+    txbuff1[1] = cfg_->PilotCi16().data();
+  }
 
   std::vector<std::vector<std::complex<int16_t>>> buff;
   // int ant = cfg_->num_channels();
-  size_t m = cfg_->NumAntennas();
+  size_t m = cfg_->BsAntNum();
   size_t r = cfg_->NumRadios();
-  // TODO: Support 2-channels
-  assert(m == r);
-  buff.resize(m * m);
-  for (size_t i = 0; i < m; i++) {
-    for (size_t j = 0; j < m; j++) {
-      if (i == j) {
-        buff[i * m + j] = cfg_->PilotCi16();
-      } else {
-        buff[i * m + j].resize(read_len);
-      }
-    }
+  size_t ref = cfg_->RefAnt() / cfg_->NumChannels();
+  buff.resize(2 * m);  // allocate for uplink and downlink directions
+  for (size_t i = 0; i < 2 * m; i++) {
+    buff[i].resize(read_len);
   }
 
   std::vector<std::complex<int16_t>> dummybuff(read_len);
   DrainBuffers();
 
   for (size_t i = 0; i < r; i++) {
-    ba_stn_[i]->setGain(SOAPY_SDR_TX, ch, "PAD",
-                        ch != 0u ? cfg_->CalibTxGainB() : cfg_->CalibTxGainA());
+    for (size_t ch = 0; ch < cfg_->NumChannels(); ch++) {
+      ba_stn_[i]->setGain(
+          SOAPY_SDR_TX, ch, "PAD",
+          ch != 0u ? cfg_->CalibTxGainB() : cfg_->CalibTxGainA());
```

Comment:
In the future, there should probably be replaced by a function call in the config class and limit access to CalibTxGainB / CalibTxGainA.   cfg_->GetCalibTxGain( channel ).  Same with the rx gains, and non calib.  Probably encapsulate all variables that go with a channel into a class.  Then if we ever need more than 2, it should be simple.